### PR TITLE
Rename reduceAsync() to foldAsync()

### PR DESCRIPTION
**Pull request by de1ux**
Quick CMD + SHIFT + R to close out #241

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Utilities for working with Futures, Streams and async computations.
 `enumerate` and `concat` represent `Stream` versions of the same-named
 [quiver.iterables][] methods.
 
-`doWhileAsync`, `reduceAsync` and `forEachAsync` perform async computations on
+`doWhileAsync`, `foldAsnyc` and `forEachAsync` perform async computations on
```
Comment (hterkelsen) [inline README.md]: s/foldAsnyc/foldAsync/

File: test/async/iteration_test.dart [hunk 2]
```diff
@@ -44,10 +44,10 @@ main() {
     });
   });
 
-  group('reduceAsync', () {
+  group('foldAsync', () {
     test('should reduce iterable', () {
```
Comment (hterkelsen) [inline test/async/iteration_test.dart]: 'should fold iterable'

File: README.md (hunk 1 above)
Comment (de1ux) [inline README.md]: bah, thanks

**Comment by yjbanov** [comment]
This is a good fix, but breaking. Let's fix it in https://github.com/QuiverDart/quiver_async and keep this one broken for backwards-compatibility.

**Pull request closed without merging**